practice/earthquake/site.js

```diff
@@ -0,0 +1,170 @@
+ * coordinates.
+ */
+function formatSingleEarthquake(earthquake) {
+  var earthquakeProperies = earthquake.properties;
+  var earthquakeGeometry = earthquake.geometry;
+  var earthquakeTime = earthquakeProperies.time;
+  var earthquakeAgeHours =
+    (new Date().getTime() - earthquakeTime) / (1000 * 60 * 60);
+  var earthquakeAgeHoursClean = earthquakeAgeHours.toFixed(2);
+  var earthquake = {};
+  if (earthquakeProperies.mag > 0) {
+    earthquake.magnitude = earthquakeProperies.mag;
+  } else {
+    earthquake.magnitude = .001;
+  }
+  earthquake.ageInHours = earthquakeAgeHoursClean;
+  earthquake.XYCoords = earthquakeGeometry.coordinates.slice(0, 2);
+  return earthquake;
+}
+
+/**
+ * Takes in the earthquakes given from the USGS and passes each item into the
+ * format single earthquake function (above) as part of a mapping, returning an
+ * arry of simple earthquakes.
+ */
+function formatEarthquakes(earthquakes) {
+  var earthquakeFeatures = earthquakes.features;
+  var earthquakesAsMagnitutdeAgeCoords = _.map(
+    earthquakeFeatures, formatSingleEarthquake
+  );
+  return earthquakesAsMagnitutdeAgeCoords;
+}
+
+
+
+/**
+ * Creates an array of vector layers to be added to the map variable, by passing
+ * in a list of simple earthquakes, and assigning layer values based on the
+ * magnitude, age, and location of the earthquake.
+ *
+ * Earth renders fine, but dots don't show up at all... I'm kind of at a loss on
+ * what's going wrong.
+ *
+ */
+// function getDots(earthquakes) {
+//   var dots = [earth];
+//   for (var i = 0; i < earthquakes.length; i += 1) {
+//     var layer = new ol.layer.Vector({
+//       source: new ol.source.Vector({
+//         features: new ol.Feature({
+//           'geometry': new ol.geom.Point(
+//             earthquakes[i].XYCoords
+//           )
+//         })
+//       }),
+//       //   })
+//       // })
+//     });
+//     dots.push(layer);
+//   }
+//   return dots;
+// }
+
+
+/**
+ * This function formats the eathquake information into simple earthquakes, then
+ * creates a series of vector layers with getDots, and passes them all into a
+ * map with an additional earth layer.
+ */
+function renderEarthquakes(earthquakes) {
+  var simpleEarthquakes = formatEarthquakes(earthquakes);
+  // var dots = getDots(simpleEarthquakes);
+  // console.dir(dots);
+  $('.map').empty();
+  // var map = new ol.Map({
+  //   layers: dots,
+  //   target: 'map',
+  //   view: new ol.View({
+  //     center: [0, 0],
+  //     zoom: 2})
+  var count = simpleEarthquakes.length;
+  var features = new Array(count);
+  for (var i = 0; i < count; ++i) {
+    features[i] = new ol.Feature({
+      'geometry': new ol.geom.Point(
+        ol.proj.fromLonLat(
+          [simpleEarthquakes[i].XYCoords[0],
+          simpleEarthquakes[i].XYCoords[1]])
+        ),
+      'size': simpleEarthquakes[i].magnitude,
+      'opacity': 1 - (simpleEarthquakes[i].ageInHours / 168),
+    });
+  }
```

Comment:
This is a classic mapping. If you use `_.map` you don't need to coordinate array lengths, counting, subscripting, etc.